test/common/stats/thread_local_store_speed_test.cc

```diff
@@ -37,9 +37,11 @@ class ThreadLocalStorePerf {
     for (auto& stat_name_storage : stat_names_) {
       stat_name_storage->free(symbol_table_);
     }
-    store_.shutdownThreading();
```

Comment:
What's the purpose of this change?

Reply:
In production code we were shutting down threading this way, where ThraedLocal::shutdownGlobalThreading() was called first. That means in shutdownThreading() we can't then go through the runOAllThreads sequence to clean up; we just have to remove the elements directly.

This change, and others in some tests, are to make tests shutdown threading in the same order as production code. The `ASSERT(tls_->isShutdown())`, validates the claim made below it:
```
// We can't call runOnAllThreads here as global threading has already been shutdown.
```

and a lot of the trivial test changes are to avoid having that assert crash in unit tests.